admin/python/lsst/qserv/testing/README.md

```diff
@@ -0,0 +1,164 @@
+
+# Qserv KPM test harness
+
+This package contains Python code and configuration examples for PM testing of
+QServ cluster. Main application is `qserv-kraken` (lives in `admin/bin/`) and
+it uses the code in this package. Folder `config` in this package contains an
+example of the configuration file that can be used for running `qserv-kraken`.
+
+`qserv-kraken` runs multiple "clients", each client generating queries of
+specific query class and executing them sequentially. Total number of clients
+is determined by configuration (sum of all `concurrentQueries` values), and it
+is typically around 100. Every client in this implementation runs in a
+separate sub-process, all those sub-processes are started and managed by
+`qserv-kraken`. Depending on hardware it may be better to run the whole
+shebang on several hosts, `qserv-kraken` provides simple way to do that.
+
+
+## Configuration
+
+The queries that need to be executed, number of clients and few other
+parameters are specified in the YAML configuration file, here is the
+description of its contents.
+
+At the top level config file has two keys:
+
+- `queryClasses` -- defines execution policies for different classes of
+  queries,
+- `queries` -- defines queries to be executed.
+
+For each of these two keys the next level defines a bunch of keys
+corresponding to query classes ("LV", "NearN", etc.)
+
+Below that `queryClasses` tree defines parameters for query execution
+with these keys:
+
+- `concurrentQueries` -- (`int`) number of concurrent queries to run, which is
+  also a number of sub-processes that will be executing them;
+- `targetTime` -- target time for query execution, this is not used currently;
+- `maxRate` -- (`float`, optional) maximum query submission rate in Hz for
+  queries of this class for single client, if not specified or `null` then
+  rate is not limited;
+- `arraysize` -- (`int`, optional) number of result rows to fetch at once with
+  `fetchmany()` DBAPI call, if not specified or `null` then driver default is
+  used.
+
+For `queries` tree the level below query classes contain query definitions.
```

Comment:
to me, "level below" makes it sound like queries is within queryClasses, but around line 26 it sounds like query and queryClasses are peers. I'm not sure which is right or if I'm misunderstanding in general, but I'm confused by this wording.

Reply:
"queries" and "queryClasses" are two separate configuration trees at the top level, but each of them have query classes as keys at one level. I'll try to fix poor wording.